### PR TITLE
Implemented address vectors and did some fixes for OpenCV support.

### PR DESCRIPTION
**Pull request by rmosaner**
The majority of changes tackles address vector support. Besides that, I implemented some built-ins which are needed in OpenCV and did some minor changes for particular OpenCV incompatibilities.

### CODE REVIEW

File: projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/types/VectorType.java [hunk 1]
```diff
@@ -33,15 +33,24 @@
...
         this.length = length;
     }
 
-    public PrimitiveType getElementType() {
+    public VectorType(PointerType elementType, int length) {
+        this.elementType = elementType;
+        this.length = length;
+    }
+
+// public PrimitiveType getElementType() {
```
Comment (jkreindl) [inline projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/types/VectorType.java]: This code should be removed, not just commented out.

**Comment by jkreindl** [comment]
LGTM!

File: projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/types/VectorType.java (hunk 1 above)
Comment (rmosaner) [inline projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/types/VectorType.java]: This comment was already removed in the second commit. You maybe have looked only at the first of the four commits.

Comment (jkreindl) [inline projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/types/VectorType.java]: Yes, I did and I already noticed this myself. Sorry about my unneccessary comment.

**Pull request merged**